pkg/util/log/slog/wrapper.go

```diff
@@ -0,0 +1,215 @@
+}
+
+// Trace formats message using the default formats for its operands
+// and writes to log with level = Trace
+func (w *Wrapper) Trace(v ...interface{}) {
+	w.handleArgs(types.TraceLvl, v...)
+}
+
+// Tracef formats message according to format specifier
+// and writes to log with level = Trace.
+func (w *Wrapper) Tracef(format string, params ...interface{}) {
+	w.handleFormat(types.TraceLvl, format, params...)
+}
+
+// Debug formats message using the default formats for its operands
+// and writes to log with level = Debug
+func (w *Wrapper) Debug(v ...interface{}) {
+	w.handleArgs(types.DebugLvl, v...)
+}
+
+// Debugf formats message according to format specifier
+// and writes to log with level = Debug.
+func (w *Wrapper) Debugf(format string, params ...interface{}) {
+	w.handleFormat(types.DebugLvl, format, params...)
+}
+
+// Info formats message using the default formats for its operands
+// and writes to log with level = Info
+func (w *Wrapper) Info(v ...interface{}) {
+	w.handleArgs(types.InfoLvl, v...)
+}
+
+// Infof formats message according to format specifier
+// and writes to log with level = Info.
+func (w *Wrapper) Infof(format string, params ...interface{}) {
+	w.handleFormat(types.InfoLvl, format, params...)
+}
+
+// Warn formats message using the default formats for its operands
+// and writes to log with level = Warn
+func (w *Wrapper) Warn(v ...interface{}) error {
+	return w.handleError(types.WarnLvl, renderArgs(v...))
+}
+
+// Warnf formats message according to format specifier
+// and writes to log with level = Warn.
+func (w *Wrapper) Warnf(format string, params ...interface{}) error {
+	return w.handleError(types.WarnLvl, renderFormat(format, params...))
+}
+
+// Error formats message using the default formats for its operands
+// and writes to log with level = Error
+func (w *Wrapper) Error(v ...interface{}) error {
+	return w.handleError(types.ErrorLvl, renderArgs(v...))
+}
+
+// Errorf formats message according to format specifier
+// and writes to log with level = Error.
+func (w *Wrapper) Errorf(format string, params ...interface{}) error {
+	return w.handleError(types.ErrorLvl, renderFormat(format, params...))
+}
+
+// Critical formats message using the default formats for its operands
+// and writes to log with level = Critical
+func (w *Wrapper) Critical(v ...interface{}) error {
+	return w.handleError(types.CriticalLvl, renderArgs(v...))
+}
+
+// Criticalf formats message according to format specifier
+// and writes to log with level = Critical.
+func (w *Wrapper) Criticalf(format string, params ...interface{}) error {
+	return w.handleError(types.CriticalLvl, renderFormat(format, params...))
+}
+
+// Close flushes all the messages in the logger and closes it. It cannot be used after this operation.
+func (w *Wrapper) Close() {
+	if w.close != nil {
+		w.close()
+	}
+}
+
+// Flush flushes all the messages in the logger.
+func (w *Wrapper) Flush() {
+	if w.flush != nil {
+		w.flush()
+	}
+}
```

Comment:
there doesn't look to be a way to provide these to a new slog Wrapper publicly, my understanding is that to achieve this we would want to flush / close on the io.Writer of the slog.Handler?

or will use of this wrapper be like:

```
stdout := os.STDOUT
handler := slog.NewTextHandler(stdout, options)
loggerWrapper := NewWrapperWithCloseAndFlush(handler, stdout.Close, stdout.Flush)
```